### DISCUSSION TITLE
[iOS] Unable to debug when deploying on physical device

### DISCUSSION THREAD

**Discussion by mcocaroUX** [unoplatform/uno#15635] [Q&A] [answered]
Currently facing an issue deploying our app (also targeting Android) to an iOS physical device for debugging. Although the IDE appears to be in debug mode, it doesn't halt at any breakpoints. Additionally, we're experiencing unusual app behavior, particularly with IFeeds failing and displaying their error states.

We encountered the following exception in Main.iOS.cs:
![image](https://github.com/unoplatform/uno/assets/94198647/ecb7ac6b-c50e-45eb-8723-52035105c97e)

StackTrace: `at Uno.Extensions.PlatformHelper.EnsureInitialized()`

Interestingly, this issue doesn't occur on the iOS simulator or Android simulator and devices.

Would appreciate any insights to this problem 

Binlogs:
[binlogs.zip](https://github.com/unoplatform/uno/files/14405587/binlogs.zip)

**Comment by jeromelaban** [comment]
Thanks. This looks like a handled exception. While we'll be making changes to avoid that on iOS, you should be able to continue debugging normally.

 For the debugging, it's a visual studio behavior that don't have control over, you may want to try repairing your environment, or try using vscode with the uno platform extension instead which runs directly on the mac and is more stable.

**Reply by johnkattenhorn** [reply]
Thanks for the support @jeromelaban

**Comment by mcocaroUX** [comment, marked as answer]
@jeromelaban After running Uno checks (on windows and mac) I was still experiencing the same issue, although it fixed and updated some items. I deleted xamarin cache folder from `Users/{username}/Library/Caches/Xamarin/mtbs/{solution name}` and now I can debug.

Thanks for the help!
